File: src/test/java/com/esentri/wrabbitmq/java/TestFunctions.java
package com.esentri.wrabbitmq.java;

import com.esentri.wrabbitmq.WrabbitTopic;

import java.io.Serializable;
import java.util.UUID;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

import static org.fest.assertions.Assertions.assertThat;

public class TestFunctions {
   public static void Await(CountDownLatch latch) {
      try {
         assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
      } catch (InterruptedException e) {
         throw new RuntimeException("CountDownLatch was interrupted.");
      }
   }

   public static <MESSAGE extends Serializable, REPLY extends Serializable> WrabbitEventWithReply<MESSAGE, REPLY> NewEventWithReply(WrabbitTopic topic) {
      return new WrabbitEventWithReply<MESSAGE, REPLY>(topic, UUID.randomUUID().toString());
   }
}

File: src/test/kotlin/com/esentri/wrabbitmq/TestFunctions.kt
package com.esentri.wrabbitmq

import org.fest.assertions.Assertions.assertThat
import java.io.Serializable
import java.util.*
import java.util.concurrent.CountDownLatch
import java.util.concurrent.TimeUnit

fun Await(latch: CountDownLatch, seconds: Long = 10) {
   assertThat(latch.await(seconds, TimeUnit.SECONDS)).isTrue()
}

fun <MESSAGE : Serializable, REPLY : Serializable> NewEventWithReply(topic: WrabbitTopic) =
   WrabbitEventWithReply<MESSAGE, REPLY>(topic, UUID.randomUUID().toString())
